OADP-3604 : openshift-velero-plugin panics on imagestream backup due to a missing secret (#72456)
openshift/openshift-docs: modules/oadp-features-plugins-known-issues.adoc

// Module included in the following assemblies:
// oadp-features-plugins-known-issues

> // * backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc

Reviewer: // * backup_and_restore/application_backup_and_restore/troubleshooting.adoc

:_content-type: CONCEPT
[id="oadp-features-plugins-known-issues_{context}"]
= OADP plugins known issues

The following section describes known issues in {oadp-first} plugins:

[id="velero-plugin-panic_{context}"]

> == Velero plugin panics on imagestream backup due to a missing secret

Reviewer: The words _panics on_ sounds odd. I might be right for all I know. Would _panics during_ be correct? I think we need an article, too. 

```suggestion
== Velero plugin panics during an imagestream backup due to a missing secret
```

```suggestion
== Velero plugin panics during imagestream backups due to a missing secret
```

> The velero plugin panics on imagestream backup. A missing secret causes this panic error.

Reviewer: Do we need this sentence, as it just repeats the heading?

> When the backup and the Backup Storage Location (BSL) are managed outside the scope of the Data Protection Application (DPA), the OADP controller, meaning the DPA reconciliation does not create the relevant `oadp-<BSL Name>-registry-secret`.

Reviewer 2: ```suggestion
When the backup and the Backup Storage Location (BSL) are managed outside the scope of the Data Protection Application (DPA), the OADP controller, meaning the DPA reconciliation does not create the relevant `oadp-<BSL Name>-<BSL Provider>-registry-secret`.
```

> When the backup and the Backup Storage Location (BSL) are managed outside the scope of the Data Protection Application (DPA), the OADP controller, meaning the DPA reconciliation does not create the relevant `oadp-<BSL Name>-<BSL Provider>-registry-secret`.

Reviewer: I am not clear on what this means:   _meaning the DPA reconciliation_

Author: You reconcile a DPA by running a command such as:

```
oc get dpa dpa-sample -n openshift-adp -o jsonpath='{.status}'
```

Example output
```
{"conditions":[{"lastTransitionTime":"2023-10-27T01:23:57Z","message":"Reconcile complete","reason":"Complete","status":"True","type":"Reconciled"}]}
```

> When the backup is run, the openshift-velero-plugin panics on the imagestream backup, with a panic error:

Reviewer: I don't see `openshift-velero-plugin` in the docs Should we spell it out? [OpenShift Velero plugin](
https://docs.openshift.com/container-platform/4.15/backup_and_restore/application_backup_and_restore/oadp-features-plugins.html#default-velero-cloud-provider-plugins)

[source,terminal]
----
024-02-27T10:46:50.028951744Z time="2024-02-27T10:46:50Z" level=error msg="Error backing up item"

> backup=openshift-adp/$backupJobName error="error executing custom action (groupResource=imagestreams.image.openshift.io,

Reviewer 3: ```suggestion
backup=openshift-adp/<backup name> error="error executing custom action (groupResource=imagestreams.image.openshift.io,
```

namespace=<BSL Name>, name=postgres): rpc error: code = Aborted desc = plugin panicked:
runtime error: index out of range with length 1, stack trace: goroutine 94…
----

[id="velero-plugin-panic-workaround_{context}"]
=== Workaround to avoid the panic error

> To avoid the velero plugin panic error, follow these steps:

Reviewer: ```suggestion
To avoid the Velero plugin panic error, perform the following steps:
```

. Label the custom BSL with the relevant label
+
[source,terminal]
----

> $ oc label BackupStorageLocation <BSL Name> app.kubernetes.io/component=bsl`

Reviewer 3: ```suggestion
$ oc label BackupStorageLocation <BSL Name> app.kubernetes.io/component=bsl
```

> $ oc label BackupStorageLocation <BSL Name> app.kubernetes.io/component=bsl

Reviewer: Per the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#command-syntax-for-replaceable-values), replaceable values take an underscore between words. Also, while not in the manual, we generally use lowercase in my experience. 

```suggestion
$ oc label BackupStorageLocation <bsl_name> app.kubernetes.io/component=bsl
```

----

> . Once the BSL is labeled, wait until the DPA reconciles.

Reviewer: Per ISG.
```suggestion
. After the BSL is labeled, wait until the DPA reconciles.
```

+
[NOTE]
====

> You can force this by making any minor change to the DPA itself.

Reviewer: s/this/the reconciliation ??

```suggestion
You can force the reconciliation by making any minor change to the DPA itself.
```

====

> . When the DPA reconciles, confirm that the relevant `oadp-<BSL Name>-registry-secret` has been created and that the right registry data has been populated into it:

Reviewer 2: ```suggestion
. When the DPA reconciles, confirm that the relevant `oadp-<BSL Name>-<BSL Provider>-registry-secret` has been created and that the right registry data has been populated into it:
```

> . When the DPA reconciles, confirm that the relevant `oadp-<BSL Name>-<BSL Provider>-registry-secret` has been created and that the right registry data has been populated into it:

Reviewer: I wonder if _right_ would pose a translation issue (right/left) ??

```suggestion
. When the DPA reconciles, confirm that the relevant `oadp-<BSL Name>-<BSL Provider>-registry-secret` has been created and that the correct registry data has been populated into it:
```

+
[source,terminal]
----

> $ oc -n openshift-adp get secret/oadp-oadp-$BSL-registry-secret -o json | jq -r '.data'

Author: ```suggestion
$ oc -n openshift-adp get secret/oadp-$BSL-registry-secret -o json | jq -r '.data'
```

> $ oc -n openshift-adp get secret/oadp-<BSL Name>-registry-secret -o json | jq -r '.data'

Reviewer 2: ```suggestion
$ oc -n openshift-adp get secret/oadp-<BSL Name>-<BSL Provider>-registry-secret -o json | jq -r '.data'
```

> $ oc -n openshift-adp get secret/oadp-<BSL Name>-<BSL Provider>-registry-secret -o json | jq -r '.data'

Reviewer: ```suggestion
$ oc -n openshift-adp get secret/oadp-<bsl_name>-<bsl_provider>-registry-secret -o json | jq -r '.data'
```

The velero plugin panics on imagestream backup. A missing secret causes this panic error.

When the backup and the Backup Storage Location (BSL) are managed outside the scope of the Data Protection Application (DPA), the OADP controller, meaning the DPA reconciliation does not create the relevant `oadp-<backup storage location name>-registry-secret`.

When the backup is run, the openshift-velero-plugin panics on the imagestream backup, with a panic error:

[source,terminal]
----
024-02-27T10:46:50.028951744Z time="2024-02-27T10:46:50Z" level=error msg="Error backing up item"
backup=openshift-adp/$backupJobName error="error executing custom action (groupResource=imagestreams.image.openshift.io,

> namespace=<backup storage location name>, name=postgres): rpc error: code = Aborted desc = plugin panicked:

Reviewer 3: don;t kill me
<BSL Name>

Author: `<backup storage location name>` changed to `<BSL Name>`
